src/lib/hooks/hasConnectionChange.js

```diff
@@ -46,54 +40,63 @@ export const useConnectionWeb3 = () => {
           .then(_ => true)
           .catch(_ => false))
       ) {
-        if (needToBindEventsWeb3) {
-          bindEvents('add')
-          needToBindEventsWeb3 = false
-        }
+        log.debug('web3 settings connection back')
         setIsConnection(true)
       } else {
         //if not connected and not reconnecting than try to force reconnect
         if (wallet.wallet.currentProvider.reconnecting === false) {
           wallet.wallet.currentProvider.reconnect()
         }
+        log.debug('web3 settings connection lost')
         setIsConnection(false)
-        if (!needToBindEventsWeb3) {
-          needToBindEventsWeb3 = true
-          bindEvents('remove')
-        }
-        setTimeout(isWeb3Connection, 1000)
       }
     } else {
       setIsConnection(true)
     }
-  }
+  }, [setIsConnection, wallet])
 
   const web3Close = useCallback(() => {
     log.debug('web3 close')
     isWeb3Connection()
-  }, [])
+  }, [isWeb3Connection])
 
   const web3Error = useCallback(() => {
     log.debug('web3 error')
     isWeb3Connection()
-  }, [])
-
-  const bindEvents = method => {
-    log.debug('web3 binding listeners')
-    const callMethod = method === 'remove' ? 'off' : 'on'
-    wallet.wallet.currentProvider[callMethod]('close', web3Close)
-    wallet.wallet.currentProvider[callMethod]('error', web3Error)
-  }
+  }, [isWeb3Connection])
 
   useEffect(() => {
-    if (wallet && appState === 'active') {
-      isWeb3Connection()
+    const bindEvents = method => {
```

Comment:
Those event listeners also should be removed in the cancel hook
Similar case i've found in the useConnectionGun also

Reply:
fixed